src/services/pscExtensionService.ts

```diff
@@ -59,3 +59,73 @@ export const createPscExtension = async (request: Request, transactionId: string
 
     return castedSdkResponse;
 };
+
+export const getIsPscExtensionValid = async (request: Request, transactionId: string, pscNotificationId: string, companyNumber: string): Promise<ValidationStatusResponse> => {
+    if (!transactionId) {
+        throw new Error(`Aborting: transactionId is required for PSC Extension validation GET request`);
+    }
+    if (!pscNotificationId) {
+        throw new Error(`Aborting: pscNotificationId is required for PSC Extension validation GET request`);
+    }
+    if (!companyNumber) {
+        throw new Error(`Aborting: companyNumber is required for PSC Extension validation GET request`);
+    }
```

Comment:
```suggestion
    const params = { transactionId, pscNotificationId, companyNumber };
    if (!Object.values(params).every(Boolean)) {
      const missing = Object.keys(params).filter(key => !params[key]);
      throw new Error(`Aborting: Missing required parameters: ${missing.join(', ')}`);
    }
```

The refactored version validates all required parameters at once and reports all missing ones in a single error message